plugins/plugin-cpp/che-plugin-cpp-lang-server/src/main/java/org/eclipse/che/plugin/cpp/generator/CProjectGenerator.java

```diff
@@ -10,33 +10,41 @@
  */
 package org.eclipse.che.plugin.cpp.generator;
 
-import com.google.inject.Inject;
+import static org.eclipse.che.api.fs.server.WsPathUtils.resolve;
+
+import java.io.InputStream;
 import java.util.Map;
+import javax.inject.Inject;
 import org.eclipse.che.api.core.ConflictException;
 import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
 import org.eclipse.che.api.core.ServerException;
-import org.eclipse.che.api.project.server.FolderEntry;
+import org.eclipse.che.api.fs.server.FsManager;
 import org.eclipse.che.api.project.server.handlers.CreateProjectHandler;
 import org.eclipse.che.api.project.server.type.AttributeValue;
-import org.eclipse.che.api.vfs.Path;
-import org.eclipse.che.api.vfs.VirtualFileSystem;
-import org.eclipse.che.api.vfs.VirtualFileSystemProvider;
 import org.eclipse.che.plugin.cpp.shared.Constants;
 
 public class CProjectGenerator implements CreateProjectHandler {
 
-  @Inject private VirtualFileSystemProvider virtualFileSystemProvider;
-
   private static final String FILE_NAME = "hello.c";
+  private static final String RESOURCE_NAME = "files/default_c_content";
+
+  private final FsManager fsManager;
+
+  @Inject
+  public CProjectGenerator(FsManager fsManager) {
+    this.fsManager = fsManager;
+  }
 
   @Override
   public void onCreateProject(
-      Path projectPath, Map<String, AttributeValue> attributes, Map<String, String> options)
-      throws ForbiddenException, ConflictException, ServerException {
-    VirtualFileSystem vfs = virtualFileSystemProvider.getVirtualFileSystem();
-    FolderEntry baseFolder = new FolderEntry(vfs.getRoot().createFolder(projectPath.toString()));
-    baseFolder.createFile(
-        FILE_NAME, getClass().getClassLoader().getResourceAsStream("files/default_c_content"));
+      String projectWsPath, Map<String, AttributeValue> attributes, Map<String, String> options)
+      throws ForbiddenException, ConflictException, ServerException, NotFoundException {
+
+    fsManager.createDir(projectWsPath);
+    InputStream inputStream = getClass().getClassLoader().getResourceAsStream(RESOURCE_NAME);
+    String wsPath = resolve(projectWsPath, FILE_NAME);
+    fsManager.createFile(wsPath, inputStream);
```

Comment:
do we need to close inputStream here?

Reply:
Again, I'm not sure if we need to close the stream here, probably we do, however this is out of scope of this issue. The PR already has like three hundreds of file changed, honestly, I see those and some other errors, but I haven't the slightest desire to mix those fixes with this huge PR. I'm sure that there will be enough PR-related bugs and errors.

If you are really concerned about such kind of errors (and I sincerely share your concern) I guess you can create a dedicated issue so we could later get back to it and fix all those bugs.

Reply:
Created an issue: https://github.com/eclipse/che/issues/6841.